### PR TITLE
Update Windows build and remove MX support from build system

### PR DESCRIPTION
**Pull request by dudamoos**
This updates the Visual Studio project files to remove the MX configuration that is not longer supported in GLEW 2.0. It also completely removes all dependencies on the C standard library for release builds on Windows. This should make it easier to use GLEW with whatever C compiler you choose since there will be no C library conflicts (later versions of MSVC add in lots of security stuff that depends on the C library but which aren't needed in GLEW).

### CODE REVIEW

**Comment by nigels-com** [comment]
Overall, this is all good progress, thank you.

I do have a reservation about bumping the required cmake version and having the additional compilation unit.  Hopefully we can #ifdef that into glew.c and not depend on cmake complexity to solve the problem.  I also need to check that the native make build works too.  (cmake to me is still the secondary build system)

**Comment by dudamoos** [comment]
The reason for removing the MX configurations from the build scripts (CMake, Visual Studio, and Makefiles) is because you said MX is no longer supported, so there shouldn't be options to build it. As for removing GLEW's dependency on the standard library, that seems to have been the original goal (as evidenced by all the functions in glew.c that replace functions from the standard library). Additionally, on my machine removing all the standard library stuff and using nostdlib/nodefaultlib reduces the size of the library by about half with no loss in functionality.

When you reminded me of the Makefile build, I went through and ripped MX support out of those build files too, since it's no longer supported. That's a lot of changes though, so I might split this into 2 pull requests, though I'd need to figure out how to split some of the commits into the part that removes MX support and the part that removes the dependency on the standard library. Which way would you prefer?

**Comment by nigels-com** [comment]
Reducing the size of the library by 50% is an excellent outcome, thank you!

-- Nigel

File: config/Makefile.linux-mingw64 [hunk 1]
```diff
@@ -6,11 +6,11 @@
 
 NAME  := glew32
 HOST  := i686-w64-mingw32
-CC    := $(HOST)-gcc
-LD    := $(HOST)-ld
+CC    := $(HOST)-gcc -fno-builtin -fno-stack-protector
```
Comment (nigels-com) [inline config/Makefile.linux-mingw64]: I'd prefer these flags moved to _CFLAGS.EXTRA_ so that overriding CC and LD doesn't result in these being lost.

**Comment by nigels-com** [comment]
I was thinking of the possibility of adding MX support back in, depending on the reaction.  From that point of view, it would be nice to have the MX untangled from the standard library dependency.  But honestly, it's only resolving the cmake version that's important to me.

**Comment by nigels-com** [comment]
Merged, thanks.

**Pull request merged**